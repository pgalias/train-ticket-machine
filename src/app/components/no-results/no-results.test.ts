import { shallowMount } from '@vue/test-utils';
import NoResults from './no-results.vue';

describe('NoResults', () => {
  test('should render component with default message when message prop was not passed', () => {
    // @ts-ignore
    const wrapper = shallowMount(NoResults);

    expect(wrapper.html()).toMatchSnapshot();
  });

  test('should render component with passed message', () => {
    // @ts-ignore
    const wrapper = shallowMount(NoResults, {
      propsData: {
        message: 'Not any foo found',
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  test('should render component with message without emoji when showEmoji prop is false', () => {
    // @ts-ignore
    const wrapper = shallowMount(NoResults, {
      propsData: {
        message: 'Not any foo found',
        showEmoji: false,
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
