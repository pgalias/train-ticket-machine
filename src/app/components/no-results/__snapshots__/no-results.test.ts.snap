// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoResults should render component with default message when message prop was not passed 1`] = `<p class="no-results">No results found <img class="emoji" alt="sad face emoji"></p>`;

exports[`NoResults should render component with message without emoji when showEmoji prop is false 1`] = `
<p class="no-results">Not any foo found
  <!--v-if-->
</p>
`;

exports[`NoResults should render component with passed message 1`] = `<p class="no-results">Not any foo found <img class="emoji" alt="sad face emoji"></p>`;
